app/src/main/java/com/simpragma/recipe/util/SearchItemActivity.java

```diff
@@ -0,0 +1,294 @@
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.StringRequest;
+import com.android.volley.toolbox.Volley;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.simpragma.recipe.Constants;
+import com.simpragma.recipe.recipeapp.R;
+import com.simpragma.recipe.roomDatabase.AppDatabase;
+import com.simpragma.recipe.adapterclass.DataAdapter;
+import com.simpragma.recipe.pojoclass.Recipe;
+import com.simpragma.recipe.roomDatabase.RecipeRoomDB;
+import com.simpragma.recipe.pojoclass.RecipeList;
+import com.simpragma.recipe.roomDatabase.RoomDatabaseAdapter;
+
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class SearchItemActivity extends AppCompatActivity implements View.OnClickListener {
+
+    EditText searchEditText;
+    Button searchButton;
+    String TAG = getClass().getSimpleName(), url;
+    ArrayList<RecipeList> reciperesults;
+    ArrayList<RecipeList> addRecipelist;
+    Recipe post;
+    RecyclerView recyclerView;
+
+    GridLayoutManager gridLayoutManager;
+    ProgressBar progressBar;
+
+    private final int PAGE_START = 1;
+    private boolean isLoading = false;
+    private int TOTAL_PAGES = 3;
+    private int currentPage;
+
+    private boolean loading = true;
+    int pastVisiblesItems, visibleItemCount, totalItemCount;
+
+    private DataAdapter customAdapter;
+    TextView textView;
+    AppDatabase roomDB;
+    RecipeRoomDB recipeRoomDB;
+    List<com.simpragma.recipe.roomDatabase.RecipeRoomDB> recipesList;
+    RoomDatabaseAdapter roomDataBaseAdapter;
+    String version = "";
+    Set<RecipeList> set;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_search_item1);
+
+        if (com.simpragma.recipe.Constants.type == Constants.Type.Production) {
+            version = checkPackageInfo();
+            Log.d(TAG, " PRODUCTION VERSION" + version);
+        } else {
+            version = checkPackageInfo();
+            Log.d(TAG, " STAGING VERSION" + version);
+        }
+
+        searchEditText = (EditText) findViewById(R.id.et_search);
+        searchButton = (Button) findViewById(R.id.bt_search);
+        textView = (TextView) findViewById(R.id.tv_notfound);
+        addRecipelist = new ArrayList<RecipeList>();
+        reciperesults = new ArrayList<RecipeList>();
+        set = new HashSet<RecipeList>();
+
+        recyclerView = (RecyclerView) findViewById(R.id.card_recycler_view);
+        gridLayoutManager = new GridLayoutManager(this, 2);
+        recyclerView.setLayoutManager(gridLayoutManager);
+
+        //Declare Room database object
+        roomDB = Room.databaseBuilder(getApplicationContext(),
+                AppDatabase.class, "recipeMangerRoom").allowMainThreadQueries()
```

Comment:
allowMainThreadQueries is a bad idea. DB access has to happen in async way and not in the main thread

Reply:
Removed  allowMainThreadQueries. Now using AsynckTask working in back end thread

Reply:
Removed allowMainThreadQueries. All queries running in AsynckTask thread.